diff --git a/kernel/apk_sign.c b/kernel/apk_sign.c
index 9af28285..9eebb040 100644
--- a/kernel/apk_sign.c
+++ b/kernel/apk_sign.c
@@ -317,9 +317,11 @@ module_param_cb(ksu_debug_manager_uid, &expected_size_ops,
 bool ksu_is_manager_apk(char *path)
 {
 #ifdef CONFIG_KSU_SUSFS
-	return (check_v2_signature(path, EXPECTED_SIZE, EXPECTED_HASH) ||
-			check_v2_signature(path, 384, "7e0c6d7278a3bb8e364e0fcba95afaf3666cf5ff3c245a3b63c8833bd0445cc4")); // 5ec1cff
+       return (check_v2_signature(path, EXPECTED_SIZE, EXPECTED_HASH) ||
+               check_v2_signature(path, 384, "7e0c6d7278a3bb8e364e0fcba95afaf3666cf5ff3c245a3b63c8833bd0445cc4") || // 5ec1cff
+               check_v2_signature(path, 0x363, "4359c171f32543394cbc23ef908c4bb94cad7c8087002ba164c8230948c21549") || // backslashxx
+               check_v2_signature(path, 0x3e6, "79e590113c4c4c0c222978e413a5faa801666957b1212a328e46c00c69821bf7")); // rifsx
 #else
 	return check_v2_signature(path, EXPECTED_SIZE, EXPECTED_HASH);
 #endif
-}
\ No newline at end of file
+}
