From d744c4757eab6a106a51f4e55528b442ed14916d Mon Sep 17 00:00:00 2001
From: "Christopher A. Williamson" <home@chrisaw.com>
Date: Sat, 22 Jun 2024 21:38:22 +0100
Subject: [PATCH 1/2] Disable defconfig check

---
 build.config.gki | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/build.config.gki b/build.config.gki
index 4b931d9eb..1bf34d3c5 100644
--- a/build.config.gki
+++ b/build.config.gki
@@ -1,2 +1,2 @@
 DEFCONFIG=gki_defconfig
-POST_DEFCONFIG_CMDS="check_defconfig"
+POST_DEFCONFIG_CMDS=""
-- 
2.44.1
